src/bp/nlu-core/clustering.ts

```diff
@@ -0,0 +1,67 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import nearestVector from 'ml-nearest-vector'
+
+import { euclideanDistanceSquared } from './tools/math'
+import { Intent, SerializedKmeansResult, Tools } from './typings'
+import Utterance, { UtteranceToken } from './utterance/utterance'
+
+const NUM_CLUSTERS = 8
+const KMEANS_OPTIONS = <sdk.MLToolkit.KMeans.KMeansOptions>{
+  iterations: 250,
+  initialization: 'random',
+  seed: 666, // so training is consistent
```

Comment:
I know this is OOS of this PR but can you use NLU seed instead

Reply:
:+1:

Reply:
will do in another PR actually